### ISSUE TITLE
Suggestion: Use Sonar inspired test tooling

### ISSUE THREAD

**Issue by Corniel** [meziantou/Meziantou.Analyzer#415]
First of all: nice analyzer(s)! [Sonar](https://github.com/SonarSource/sonar-dotnet) has a big set of really good rules.

I'd tried to read your test scenario's. I thought they might become easier to follow with a fork I made of Sonar's test tooling (they don't ship that part of their solution): [Roslyn test-tools](https://github.com/Corniel/roslyn-test-tools).

I use it for my (way smaller) set of [analyzers](https://github.com/Qowaiv/qowaiv-analyzers).

**Comment by meziantou** [comment]
If you want to change the test framework in this project, it should be for the official test framework (all the `Microsoft.CodeAnalysis.*.Testing.XUnit` package).

**Comment by Corniel** [comment]
That does not have the benefits from what I proposed. :)

The point is that my suggestion allows to define within the the snippet or code file, you can define what a certain analyzer should find. Like this:

``` C#
using System;

public class DateTimeAsProvider
{
    public void Noncompliant()
    {
        var now = DateTime.Now; // Noncompliant {{Use a testable (date) time provider instead.}}
        //        ^^^^^^^^^^^^
        var utc = DateTime.UtcNow; // Noncompliant
        var today = DateTime.Today; // Noncompliant
    }

    public void CompliantAre()
    {
        var other = DateTime.DaysInMonth(2000, 2); // Compliant
        var noSystemDateTime = NotSystem.DateTime.Now; // Compliant
    }
}
```
